### PR TITLE
fix(typescript): bump version to fix dependency issue and topology build

### LINKED ISSUE #3988: Bug - incompatibility between typedoc version and typescript version
**Describe the problem**
A clear and concise description of the problem. Which pages are affected?
Typedoc 0.23 has [dropped support](https://github.com/TypeStrong/typedoc/issues/1945#issuecomment-1143061412) for any Typescript version < 4.6.

See installation requirements [here](https://typedoc.org/guides/installation/#requirements).

Topology also lists Typedoc as a dependency and the [build fails](https://github.com/patternfly/react-topology/issues/183) since we are pulling in these conflicting versions from the docs framework.

We'll have to bump the Typescript version up to at least 4.6 (topology and [other extensions](https://github.com/patternfly/react-console/blob/main/package.json#L54) use 4.7.4).

**Expected behavior**
A clear and concise description of the expected behavior.
Topology build succeeds with latest v6 alphas.

### PR DESCRIPTION
**Pull request by jenny-s51**
Closes #3988

### CODE REVIEW

File: packages/documentation-framework/package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "style-to-object": "0.3.0",
     "to-vfile": "6.1.0",
     "typedoc": "0.23.0",
-    "typescript": "4.3.5",
+    "typescript": "4.7.4",
```
Comment (dlabaj) [inline packages/documentation-framework/package.json:63]: This update looks fine to me.  However long term I wondering if most of the dependencies should be moved to dev and peer.  @wise-king-sullyman Do you see any issues with moving these dependencies in the future?

**Review by dlabaj — APPROVED** [review]
LGTM

File: packages/documentation-framework/package.json (hunk 1 above)
Comment (wise-king-sullyman) [reply packages/documentation-framework/package.json:63]: Yeah I'm not sure how many should be moved, but it is probably something that should be investigated.

**Review by wise-king-sullyman — APPROVED** [review]

**Review by jessiehuff — APPROVED** [review]

**Review by jenny-s51 — COMMENTED** [review]
There seems to be a parsing issue that's causing the build error... investigating...

**Comment by jenny-s51** [comment]
Closing in favor of #4004

**Pull request closed without merging**